node/src/main/scala/com/wavesplatform/transaction/TxValidationError.scala

```diff
@@ -47,94 +48,89 @@ object TxValidationError {
     override def toString: String = s"InvalidSignature(${s.toString + " reason: " + details})"
   }
 
+  sealed trait WithLog extends Product with Serializable {
+    def log: Log[Id]
+  }
+
   /** Errors which can produce failed transaction */
-  sealed trait FailedTransactionError extends Product with Serializable with ValidationError {
+  case class FailedTransactionError private (
```

Comment:
Marking case class constructor as private does not prevent others from creating new instances by calling companion object's `apply` method. Besides, Is there really a reason to make the constructor private?